v2/tools/generator/internal/astmodel/resource_type.go

```diff
@@ -596,13 +605,31 @@ func (resource *ResourceType) generateMethodDecls(codeGenerationContext *CodeGen
 	var result []dst.Decl
 
 	for _, f := range resource.Functions() {
-		funcDef := f.AsFunc(codeGenerationContext, typeName)
+		funcDef := resource.generateMethodDeclForFunction(codeGenerationContext, typeName, f)
 		result = append(result, funcDef)
 	}
 
 	return result
 }
 
+func (resource *ResourceType) generateMethodDeclForFunction(
```

Comment:
Since we're not using any values from the receiver type, can consider putting this method somewhere in common to be re-used by ObjectType as well.

Reply:
Done.